pandas/tests/arrays/datetimes/test_constructors.py

```diff
@@ -232,9 +232,10 @@ def test_from_arrowtest_from_arrow_with_different_units_and_timezones_with_(
     dtype = DatetimeTZDtype(unit=pd_unit, tz=pd_tz)
 
     result = dtype.__from_arrow__(arr)
-    expected = DatetimeArray._from_sequence(
-        np.array(data, dtype=f"datetime64[{pa_unit}]").astype(f"datetime64[{pd_unit}]"),
-        dtype=dtype,
+    expected = (
+        DatetimeArray._from_sequence(data, dtype=f"datetime64[{pa_unit}]")
+        .tz_localize("UTC")
```

Comment:
1) we can actually do better than my previous suggestion: `_from_sequence(int64_data, dtype="M8[unit, UTC]").astype(...`) avoids a copy in tz_localize.
2) better to do this construction in the non-test place and use simple_new in the test.

Reply:
With "in the non-test place", you mean in `__from_arrow__`? But there is tz_localize in there to avoid?

Reply:
> 1. we can actually do better than my previous suggestion: `_from_sequence(int64_data, dtype="M8[unit, UTC]").astype(...`) avoids a copy in tz_localize.

Done

> 2. better to do this construction in the non-test place and use simple_new in the test.

Not sure what you mean here either. Is it OK for me to do this in a followup?

(I'd really like to release tomorrow.)